pkg/scheduler/frameworkext/framework_extender_factory.go

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package frameworkext
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+	frameworkruntime "k8s.io/kubernetes/pkg/scheduler/framework/runtime"
+
+	koordinatorclientset "github.com/koordinator-sh/koordinator/pkg/client/clientset/versioned"
+	koordinatorinformers "github.com/koordinator-sh/koordinator/pkg/client/informers/externalversions"
+	"github.com/koordinator-sh/koordinator/pkg/scheduler/frameworkext/indexer"
+	"github.com/koordinator-sh/koordinator/pkg/scheduler/frameworkext/services"
+)
+
+var DefaultTransformers []SchedulingTransformer
+
+func RegisterDefaultTransformers(transformers ...SchedulingTransformer) {
+	DefaultTransformers = append(DefaultTransformers, transformers...)
+}
+
+type extendedHandleOptions struct {
+	servicesEngine                   *services.Engine
+	koordinatorClientSet             koordinatorclientset.Interface
+	koordinatorSharedInformerFactory koordinatorinformers.SharedInformerFactory
+	sharedListerAdapter              SharedListerAdapter
+	defaultTransformers              []SchedulingTransformer
+}
+
+type Option func(*extendedHandleOptions)
+
+func WithServicesEngine(engine *services.Engine) Option {
+	return func(options *extendedHandleOptions) {
+		options.servicesEngine = engine
+	}
+}
+
+func WithKoordinatorClientSet(koordinatorClientSet koordinatorclientset.Interface) Option {
+	return func(options *extendedHandleOptions) {
+		options.koordinatorClientSet = koordinatorClientSet
+	}
+}
+
+func WithKoordinatorSharedInformerFactory(informerFactory koordinatorinformers.SharedInformerFactory) Option {
+	return func(options *extendedHandleOptions) {
+		options.koordinatorSharedInformerFactory = informerFactory
+	}
+}
+
+func WithSharedListerFactory(adapter SharedListerAdapter) Option {
```

Comment:
No,  Abstract `SharedListerAdaptor`  to support custom different shareLister adapte strategy. And It can work with WithDefaultTransformers  to replace the default strategy. And `Option function` is  a pattern, we should be consistent.